File: javascript/src/binary-search-new.js
function chopRecursion(target, array) {
    return findInHalf(target, array, 0, array.length);
}

function findInHalf(target, array, start, end){
    var mid = parseInt( (start + end) / 2 );
    if(start < end){
        if(target < array[mid]){
            return findInHalf(target, array, start, mid);
        }
        else if(target > array[mid]){
            return findInHalf(target, array, mid+1, end);
        }
        else{
            return mid;
        }
    }
    else
        return -1;
}

File: ruby/src/binary-search-new.rb
def getResult(mid, position)
    return position == -1 ? -1 : mid+position
end

def chopRecursion(target, array)
    return -1 if array.empty? || array.size ==1 && target != array[0]
    
    mid = array.size/2
    if target < array[mid]
        return getResult(0, chopRecursion(target, array[0..mid-1]))
    elsif target > array[mid]
        return getResult(mid, chopRecursion(target, array[mid..array.size]))
    else
        return mid
    end
end

def chopIteration(target, array)
    left = 0
    mid = array.size/2
    right = array.size

    while left < right
        if target < array[mid]
            right = mid;
        elsif target > array[mid]
            left = mid + 1
        else
            return mid
        end

        mid = (left + right)/2
    end

    return -1
end
